### DISCUSSION TITLE
What is the difference between redis-stack and redis-strings implementations?

### DISCUSSION THREAD

**Discussion by flybayer** [caching-tools/next-shared-cache#478] [Q&A]
I wasn't able to find any documentation about the differences and tradeoffs.

**Comment by better-salmon** [comment]
Hey, @flybayer! You are right. Thank you for pointing it out. It is worth adding this information to the documentation.

`redis-strings` uses basic Redis functionality like `set`, `get`, `hSet`. These commands are very fast. Both keys and values are strings, so we can not search for information there. This makes it challenging to revalidate the cache on demand.

Conversely, `redis-stack` introduces new APIs like `RedisJSON` and `RedisSearch`, offering a trade-off between performance and flexibility. While it may sacrifice some speed, it empowers you to manage cache on demand more efficiently and faster. It also allows for more versatile cache usage from your application, such as creating custom revalidating strategies or generating a `sitemap.xml` without the need for a custom Handler.

Please feel free to ask any questions or suggest any ideas.

**Reply by flybayer** [reply]
Ok thanks. We are using ElastiCache Redis, and it does not support full text search so the `redis-stack` was failing.

**Reply by better-salmon** [reply]
So, give `redis-strings` a try!
